### PR TITLE
Database setup improvements

### PR DESCRIPTION
**Pull request by joshfriend**
I use a similar setup for most projects at work. It has proven to be very useful, so I thought I'd share.

TODO:

* [x] Use pytest fixture to create test database
* [x] Install Alembic
* [x] Add migrations as `post_compile` step

### CODE REVIEW

File: migrations/script.py.mako [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""${message}
```
Comment (jacebrowning) [inline migrations/script.py.mako]: How is this template used? I've never been a fan the style in generated Alembic migrations, namely:

- redundant docstring
- variables before imports

**Comment by jacebrowning** [comment]
I'm in favor of running the migrations as a post-compile hook.

File: migrations/script.py.mako (hunk 1 above)
Comment (joshfriend) [inline migrations/script.py.mako]: I did some cleanup of the template:

* Docstring now contains only the "message"
* Imports now come before the revision variables

**Comment by joshfriend** [comment]
Added migrations to `bin/post_compile`...

**Comment by joshfriend** [comment]
I don't have Heroku access to see exactly whats wrong with the deploy but I'd bet that it's with already having the schema defined by the first migration in the database already. You could fix that easily by running the following in `psql`:

```sql
CREATE TABLE alembic_version (
    version_num varchar(32) NOT NULL
);
INSERT INTO alembic_version (version_num) VALUES ('7da56a4eb387');
```

You can open a `psql` prompt by running `heroku pg:psql` in the repo root (as long as the git repo has a remote pointing to heroku).

**Comment by jacebrowning** [comment]
You should have access now.

**Comment by jacebrowning** [comment]
@joshfriend That's still failing to deploy. Does something need to be run in `app.json` for the review apps?

**Comment by jacebrowning** [comment]
I'm going to merge and see what happens...

**Pull request merged**